Installation issue: skilion-onedrive due to dmd not in $PATH

### Steps to reproduce the issue

```console
$ spack install skilion-onedrive %gcc@4.8.5
...
1 error found in build log:
     4     ==> [2019-12-17-13:13:31.845228] FILTER FILE: src/sqlite.d [replacing "std.c.stdlib"]
     5     ==> Executing phase: 'build'
     6     ==> [2019-12-17-13:13:31.846711] 'make' '-j6' 'onedrive' 'DESTDIR=/lustre/home/acct-hpc/hpc-jianwen/opt/cascadelake/linux-ce
           ntos7-haswell/gcc-4.8.5/skilion-onedrive-1.1.1-ohthifmfl67taullatpi2dixgilfs4bs' 'PREFIX=/'
     7     echo 1.1.1 >version
     8     dmd -g -ofonedrive -O -L-lcurl -L-lsqlite3 -L-ldl -J. src/config.d src/itemdb.d src/log.d src/main.d src/monitor.d src/onedr
           ive.d src/qxor.d src/selective.d src/sqlite.d src/sync.d src/upload.d src/util.d
     9     make: dmd: Command not found
  >> 10    make: *** [onedrive] Error 127
...
$ tail /tmp/hpc-jianwen/spack-stage-skilion-onedrive-1.1.1-ohthifmfl67taullatpi2dixgilfs4bs/spack-build-out.txt
make: dmd: Command not found
make: *** [onedrive] Error 127
```

### Platform and user environment

Please report your OS here:
```commandline
$ uname -a 
Linux 3.10.0-862.el7.x86_64 #1 SMP Fri Apr 20 16:44:24 UTC 2018 x86_64 x86_64 x86_64 GNU/Linux
$ lsb_release -d
Description:    CentOS Linux release 7.5.1804 (Core)
``` 

`dmd` (the reference compiler for D programming language) has been installed as dependency, located in `~/opt/cascadelake/linux-centos7-haswell/gcc-4.8.5/dmd-2.081.1-2gzzuz4ut26pj4l7om4k7lejm2r5ty6s/linux/bin64/dmd`. Location `linux/bin64` has been added to PATH in `dmd` package via:

```
run_env.prepend_path('PATH', self.prefix.linux.bin64)
```

It seems that Spack isn't aware of this when loading dmd as a dependency when building 
skilion-onedrive.

--- a/var/spack/repos/builtin/packages/dmd/package.py
+++ b/var/spack/repos/builtin/packages/dmd/package.py
@@ -37,7 +37,7 @@ class Dmd(MakefilePackage):
         env.prepend_path('LIBRARY_PATH', self.prefix.linux.lib64)
         env.prepend_path('LD_LIBRARY_PATH', self.prefix.linux.lib64)
 
-    def setup_dependent_build_environment(self, env):
+    def setup_dependent_build_environment(self, env, dependent_spec):
         self.setup_run_environment(env)
 
     def edit(self, spec, prefix):

--- a/var/spack/repos/builtin/packages/dmd/package.py
+++ b/var/spack/repos/builtin/packages/dmd/package.py
@@ -37,6 +37,9 @@ class Dmd(MakefilePackage):
         env.prepend_path('LIBRARY_PATH', self.prefix.linux.lib64)
         env.prepend_path('LD_LIBRARY_PATH', self.prefix.linux.lib64)
 
+    def setup_dependent_build_environment(self, env):
+        self.setup_run_environment(env)
+
     def edit(self, spec, prefix):
         # Move contents to dmd/
         mkdir = which('mkdir')

